src/main.js

```diff
@@ -24,6 +24,8 @@ window.addEventListener('load', function() {
 topForm.addEventListener('input', checkInputs);
 window.addEventListener('click', clickHandler);
 
+searchInput.addEventListener('input', searchIdeas);
+searchInput.addEventListener('keyup', clearSearch);
```

Comment:
This keyup might be clearing the search bar too soon. I think this will end up clearing the value after the first letter. I'd suggest an event listener on the search button, clearing the search input on click.